crates/re_query2/src/range/results.rs

```diff
@@ -0,0 +1,207 @@
+use nohash_hasher::IntMap;
+use re_log_types::{DataCell, RowId, TimeInt};
+use re_types_core::ComponentName;
+use re_types_core::{Component, DeserializationError, DeserializationResult};
+
+use crate::{Promise, PromiseResolver, PromiseResult};
+
+// ---
+
+/// Raw results for a range query.
+///
+/// The data is neither deserialized, nor resolved/converted.
+/// It it the raw [`DataCell`]s, straight from our datastore.
+///
+/// Use [`RangeResults::get`], [`RangeResults::get_required`] and [`RangeResults::get_optional`]
+/// in order to access the raw results for each individual component.
+#[derive(Debug, Clone)]
+pub struct RangeResults {
+    /// Raw results for each individual component.
+    pub components: IntMap<ComponentName, RangeComponentResults>,
+}
+
+impl Default for RangeResults {
+    #[inline]
+    fn default() -> Self {
+        Self {
+            components: Default::default(),
+        }
+    }
+}
```

Comment:
No generics -> auto-derive ⭐ 

```suggestion
#[derive(Debug, Default, Clone)]
pub struct RangeResults {
    /// Raw results for each individual component.
    pub components: IntMap<ComponentName, RangeComponentResults>,
}
```